src/bug-report/BugReportTool/BugReportTool/Main.cpp

```diff
@@ -0,0 +1,213 @@
+#include <filesystem>
+#include <fstream>
+#include <string>
+#include <vector>
+#include <Shlobj.h>
+#include <winrt/Windows.Data.Json.h>
+#include <winrt/Windows.Foundation.Collections.h>
+
+#include "ZipTools/ZipFolder.h"
+#include "../../../common/SettingsAPI/settings_helpers.h"
+
+using namespace std;
+using namespace std::filesystem;
+using namespace winrt::Windows::Data::Json;
+
+map<wstring, vector<wstring>> escapeInfo = {
+    { L"FancyZones\\app-zone-history.json", { L"app-zone-history/app-path" } },
+    { L"PowerRename\\replace-mru.json", { L"MRUList" } },
+    { L"PowerRename\\search-mru.json", { L"MRUList" } }
+};
+
+vector<wstring> filesToDelete = {
+};
+
+vector<wstring> getXpathArray(wstring xpath)
+{
+    vector<wstring> result;
+    wstring cur = L"";
+    for (auto ch : xpath)
+    {
+        if (ch == L'/')
+        {
+            result.push_back(cur);
+            cur = L"";
+            continue;
+        }
+
+        cur += ch;
+    }
+
+    if (!cur.empty())
+    {
+        result.push_back(cur);
+    }
+
+    return result;
+}
+
+void hideByXPath(IJsonValue& val, vector<wstring>& xpathArray, int p)
+{
+    if (val.ValueType() == JsonValueType::Array)
+    {
+        for (auto it : val.GetArray())
+        {
+            hideByXPath(it, xpathArray, p);    
+        }
+
+        return;
+    }
+
+    if (p == xpathArray.size() - 1)
+    {
+        if (val.ValueType() == JsonValueType::Object)
+        {
+            auto obj = val.GetObjectW();
+            if (obj.HasKey(xpathArray[p]))
+            {
+                auto privateDatavalue = JsonValue::CreateStringValue(L"<private_data>");
+                obj.SetNamedValue(xpathArray[p], privateDatavalue);
+            }
+        }
+        
+        return;
+    }
+
+    if (val.ValueType() == JsonValueType::Object)
+    {
+        IJsonValue newVal;
+        try
+        {
+            newVal = val.GetObjectW().GetNamedValue(xpathArray[p]);
+        }
+        catch (...)
+        {
+            return;
+        }
+        
+        hideByXPath(newVal, xpathArray, p + 1);    
+    }
+}
+
+void hideForFile(path dir, wstring relativePath)
+{
+    wstring jsonString, tmp;
+    dir = dir.append(relativePath);
+    wifstream file(dir);
+    while (std::getline(file, tmp))
+    {
+        jsonString += tmp;
```

Comment:
is it possible to use json::from_file here? it's in common/utils now, which is just a collection of header-only snippets w/o vcxprojs

Reply:
Yes, it is possible, added.